### ISSUE TITLE
Migrate NSDService implementation of Android to RxDnssd

### ISSUE THREAD

**Issue by thehellmaker** [balthazar/react-native-zeroconf#77]
Hi,
This is an issue to migrate NSDService implementation of android to https://github.com/andriydruk/RxDNSSD. I have used jmDNS and NSD service and they work well on new phones where as older android phones and certain versions and networks have issues. 

Explanation from the author ofRxDNSSD on why NSDService and jmdns and others are not nearly as good. 

One main advantage is rxdnssd relies on apple implementaiton of zeroconf which is the best until now and has very low fault rates. I have tested this in production in my app since a year. If you are up for this change I can pick this up and make it and send a pull request. I will anyways make this change for myself. I thought i could contribute it back as well

**Comment by balthazar** [comment]
Yeah it's true some people have had issues with android detection lately, most likely because of the new jmdns implementation. Would be great if you could shoot a PR for it, and we can direct people to use the tag and see how well it goes for them, to eventually switch over

**Comment by thehellmaker** [comment]
Absolutely. Makes sense. i will publish a PR for the same as soon as i am ready.

**Comment by Vondry** [comment]
These are really great news, thank you @thehellmaker. Could you please share your time estimation for this enhancement? :)

**Comment by eliaslecomte** [comment]
Any update @thehellmaker?

**Comment by balthazar** [comment]
Looking at the RxDNSSD API and really like what I am seeing. Not sure if @thehellmaker has time though.

@Vondry what is it that you think switching to RxDNSSD will improve?

**Comment by PhuongNguyenETIT** [comment]
https://code.google.com/archive/p/mdnsjava/source/default/source

**Comment by thehellmaker** [comment]
Hi Guys, 
Sorry didn't see the update on this thread. Let me get back to you in a day about this migration. 
1. If I plan to take this up?
2. If yes when will I have a patch for the same?

CHeers,
Akash A

**Comment by thehellmaker** [comment]
@PhuongNguyenETIT mDNS is a bad library. It is slow and secondly it doesn't work well with all android versions
Please refer to my description about how RxDNSSD is better

**Comment by thehellmaker** [comment]
@balthazar this is my first shot at native module. Could you please help me with the setup of local dependency setup for react-native-zeroconf? So is it just add dependency in package json as file://filepath ? . It seems like if i do that the module is not found

**Comment by balthazar** [comment]
Which dependency are you trying to install, and which package.json are you referring to?

**Comment by thehellmaker** [comment]
so actually i need to start working on migration of react-native-zeroconf from NSD service to rxdnssd2. For that i need to modify this native module

in my project i have a package json which says react-native-zeroconf: 'file://../react-native-zeroconf. However this always says module not found

Instead of npm remote dependency i need to 
1. clone react-native-zeroconf
2. add dependency on this locally closed module 
3. start making change and test them locally

**Comment by balthazar** [comment]
In the cloned react-native-zeroconf directory, you need to run `npm link` and then in your project run `npm link react-native-zeroconf`

**Comment by mtzfactory** [comment]
I'm very interested in this improvement! :-)

**Comment by thehellmaker** [comment]
I am complete with the implementation of this library migration. I have implemented a Zeroconf interface so that NsdService the old implmentation and DnsSd the new implementation are pluggable. Will open a pull request

**Comment by thehellmaker** [comment]
@balthazar  Please let me know if you have time to review this?

**Comment by thehellmaker** [comment]
@balthazar I do not have permission to open PR on this. Please do provide me with appropriate permissions.

**Comment by balthazar** [comment]
Anyone can PR so not sure what you want.

**Comment by thehellmaker** [comment]
In order to open pull request i need to push changes to remote branch on github and i do not have permission to push to a branch

**Comment by thehellmaker** [comment]
git --no-optional-locks -c color.branch=false -c color.diff=false -c color.status=false -c diff.mnemonicprefix=false -c core.quotepath=false -c credential.helper=sourcetree push -v --tags --set-upstream origin refs/heads/dnssd-enhancement:refs/heads/dnssd-enhancement 
Pushing to https://github.com/balthazar/react-native-zeroconf.git
remote: Permission to balthazar/react-native-zeroconf.git denied to thehellmaker.
fatal: unable to access 'https://github.com/balthazar/react-native-zeroconf.git/': The requested URL returned error: 403
Completed with errors, see above

**Comment by balthazar** [comment]
That's what a fork is for lmao

**Comment by thehellmaker** [comment]
https://github.com/balthazar/react-native-zeroconf/pull/131

**Comment by tunm1228** [comment]
> I have opened a pull request for the same #131

i get my request but it error.
bundling failed: Error: While trying to resolve module `react-native-zeroconf` from file `/***/Project.js`, the package `/***/node_modules/react-native-zeroconf/package.json` was successfully found. However, this package itself specifies a `main` module field that could not be resolved (`/*/node_modules/react-native-zeroconf/dist`. Indeed, none of these files exist:
